core/domain/fs_domain.py

```diff
@@ -108,96 +108,98 @@ def version(self):
         return self._version
 
 
-class ExplorationFileSystem(object):
-    """A datastore-backed read-write file system for a single exploration.
+class EntityFileSystem(object):
+    """A datastore-backed read-write file system for a single entity.
 
-    The conceptual intention is for each exploration to have its own asset
-    folder. An asset has no meaning outside its exploration, so the assets in
+    The conceptual intention is for each entity to have its own asset
+    folder. An asset has no meaning outside its entity, so the assets in
     these asset folders should therefore not be edited directly. They should
     only be modified as side-effects of some other operation (such as adding an
-    image to an exploration).
+    image to an entity like topics, stories or explorations).
 
-    The content of an exploration should include a reference to the asset
+    The content of an entity should include a reference to the asset
     together with the version number of the asset. This allows the
-    exploration to refer to asset versions.
+    entity to refer to asset versions.
 
     In general, assets should be retrieved only within the context of the
-    exploration that contains them, and should not be retrieved outside this
+    entity that contains them, and should not be retrieved outside this
     context.
 
     Args:
-        exploration_id: str. The id of the exploration.
+        assets_path: str. The path to the assets folder of the corresponding
+            entity. It is of the form {{entity}}/{{entity_id}}
+            (eg: exploration/{{exp_id}})
     """
 
     _DEFAULT_VERSION_NUMBER = 1
 
-    def __init__(self, exploration_id):
-        """Constructs a ExplorationFileSystem object.
+    def __init__(self, assets_path):
+        """Constructs a EntityFileSystem object.
 
         Args:
-            exploration_id: str. The id of the exploration.
+            assets_path: str. The path to the parent directory for all assets.
         """
-        self._exploration_id = exploration_id
+        self._assets_path = assets_path
```

Comment:
Perhaps we should do some initial validation that assets_path is X/Y and X is one of a predefined list, and throw an error if not?

Could also split assets_path into asset_type and asset_id (but still include validation) -- that might be better actually. We can then derive the _assets_path here.

Reply:
Yeah, will rename to entity_name and entity_id, since the  assets_path is of the form entity_name/entity_id.

Reply:
Done